cr-db/cr/db/loader.py

```diff
@@ -1,53 +1,158 @@
+'''Module for data loading and manipulation.  '''
+
+import json
 import csv
 import os
-import json
 import sys
+import numpy as np
+import ijson.backends.yajl2_cffi as ijson
 
 from cr.db.store import global_settings, connect
 
-def load_data(filename, settings=None, clear=None):
-    if settings is None:
-        settings = global_settings
-        global_settings.update(json.load(file(sys.argv[1])))
 
-    db = connect(settings)
+class Loader(object):
+    '''Class for data loading and manipulation.  '''
 
-    obj_name = os.path.basename(filename).split('.')[0]
+    @staticmethod
+    def load_data(filename, settings=None, clear=None, optimize_for='speed'):
+        '''Load multiple users from JSON into the MongoDB.'''
+        if settings is None:
+            settings = global_settings
+            global_settings.update(json.load(file(sys.argv[1])))
 
-    collection = getattr(db, obj_name)
+        db = connect(settings)
 
-    if clear:
-        collection.remove()
+        obj_name = os.path.basename(filename).split('.')[0]
 
-    with file(filename) as the_file:
-        objs = json.load(the_file)
-        for obj in objs:
-            collection.insert(obj)
+        collection = getattr(db, obj_name)
 
+        if clear:
+            collection.remove()
 
-def load_dataset(csv_filename, db):
-    with file(csv_filename, 'rU') as csv_file:
-        csv_data = csv.reader(csv_file)
-        headers = csv_data.next()
+        with file(filename) as the_file:
+            if optimize_for == 'speed':
+                objs = json.load(the_file)
+                collection.insert(objs)
```

Comment:
@percious Did an example/test with GridFS, and directly storing a file into MongoDB.